File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> track = new HashMap<>();
        for(String s: strs) {
            int[] freq = new int[26];
            for(char c: s.toCharArray()) {
                freq[c - 'a']++;
            }
            StringBuilder sb = new StringBuilder();
            for(int num: freq) {
                sb.append(num);
                sb.append("#");
            }
            track.putIfAbsent(sb.toString(),new ArrayList<>());
            track.get(sb.toString()).add(s);
        }
        return new ArrayList<>(track.values());
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        track = {}
        for i in range(len(strs)):
            freq = [0] * 26
            for j in range(0,len(strs[i])):
                index = ord(strs[i][j]) - ord('a')
                freq[index] += 1
            freqString = ""
            for j in range(0,len(freq)):
                freqString += str(freq[j])
                freqString += "#"
            if not freqString in track:
                track[freqString] = []
            track[freqString].append(strs[i])
        return list(track.values())
        